benches/transform/route.rs

```diff
@@ -0,0 +1,185 @@
+use core::fmt;
+use std::{collections::BTreeMap, time::Duration};
+
+use bytes::Bytes;
+use criterion::{
+    black_box, criterion_group, measurement::WallTime, BatchSize, BenchmarkGroup, BenchmarkId,
+    Criterion, SamplingMode, Throughput,
+};
+use value::Value;
+use vector::transforms::{
+    route::{Route, RouteConfig},
+    TransformOutputsBuf,
+};
+use vector_core::{
+    config::{DataType, Output},
+    event::{Event, EventContainer, EventMetadata, LogEvent},
+    transform::{SyncTransform, TransformContext},
+};
+
+#[derive(Debug)]
+struct Param {
+    slug: &'static str,
+    input: Event,
+    route_config: RouteConfig,
+    output_buffer: TransformOutputsBuf,
+}
+
+impl fmt::Display for Param {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.slug)
+    }
+}
+
+fn route(c: &mut Criterion) {
+    let mut group: BenchmarkGroup<WallTime> = c.benchmark_group("vector::transforms::route::Route");
+    group.sampling_mode(SamplingMode::Auto);
+
+    let mut fields = BTreeMap::new();
+    for alpha in [
+        "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r",
+        "s", "t", "u", "v", "w", "x", "y", "z",
+    ] {
+        fields.insert(String::from(alpha), Value::Bytes(Bytes::from(alpha)));
+    }
+    let event = Event::from(LogEvent::from_parts(fields, EventMetadata::default()));
+
+    let mut outputs = Vec::new();
+    for name in [
+        "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r",
+        "s", "t", "u", "v", "w", "x", "y", "z", "aa", "ba", "ca", "da", "ea", "fa", "ga", "ha",
+        "ia", "ja", "ka", "al", "ma", "na", "oa", "pa", "qa", "ra", "sa", "ta", "ua", "va", "wa",
+        "xa", "ay", "za", "ba", "bb", "bc", "bd", "be", "bf", "bg", "bh", "bi", "bj", "bk", "bl",
+        "bm", "bn", "bo", "bp", "bq", "br", "sb", "tb", "ub", "vb", "wb", "xb", "yb", "zb", "aba",
+        "bba", "bbca", "dba", "bea", "fba", "gba", "hba", "iba", "jba", "bka", "bal", "bma", "bna",
+        "boa", "bpa", "bqa", "bra", "bsa", "bta", "bua", "bva", "bwa", "xba", "aby", "zba",
+    ] {
+        outputs.push(Output {
+            port: Some(String::from(name)),
+            ty: DataType::Log,
+            log_schema_definition: None,
+        });
+    }
+    let output_buffer: TransformOutputsBuf = TransformOutputsBuf::new_with_capacity(outputs, 10);
```

Comment:
Looking at the topology it looks like it uses a capacity of whatever number of routes there are. Should we do the same here by moving this into the setup?

Reply:
Actually, I was looking at the tests 🤦

Reply:
Still not sure where this `10` comes from though, would it make sense to match whatever the topology does for this?

Reply:
Ah, it turns out this doesn't actually propagate to the OutputBuffer at all, the passed capacity. In the event we have a named output it is created like `OutputBuffer::default()`. I'll experiment with actually passing the capacity in a follow-up PR.

Reply:
10 is just an arbitrary number I typed out. I'm not sure what the topology actually does here, off the top of my head. I think, considering the nature of the transform's apparent bottlenecks, this value doesn't matter all that much.